src/Shape_buildblock/GenerateImage.cxx

```diff
@@ -300,9 +339,15 @@ save_image()
 
 shared_ptr<DiscretisedDensity<3, float>>
 GenerateImage::
-get_output_sptr()
+get_output_sptr(unsigned int frame)
 {
-  return out_density_ptr;
+  return out_density_ptr->get_density_sptr(frame);
 }
 
+shared_ptr<DynamicDiscretisedDensity>
+    GenerateImage::
+    get_all_outputs_sptr()
+{
+  return out_density_ptr;
+}
```

Comment:
it would be, or actually we wouldn't really need `get_output_sptr(frame)` really. However, sadly I cannot see a way to preserve backwards compatibility.